README
NAME

    Dist::Zilla::PluginBundle::Author::Plicease - Dist::Zilla plugin bundle
    used by Plicease

VERSION

    version 1.62

SYNOPSIS

    In your dist.ini:

     [@Author::Plicease]

DESCRIPTION

    This Dist::Zilla plugin bundle is mostly equivalent to

     # Basic - UploadToCPAN, Readme, ExtraTests, and ConfirmRelease
     [GatherDir]
     exclude_filename = Makefile.PL
     exclude_filename = Build.PL
     exclude_filename = cpanfile
     exclude_match    = ^_build/
     [PruneCruft]
     except = .travis.yml
     [ManifestSkip]
     [MetaYAML]
     [License]
     [ExecDir]
     [ShareDir]
     [MakeMaker]
     [Manifest]
     [TestRelease]
     
     [Author::Plicease::PrePodWeaver]
     [PodWeaver]
     [NextRelease]
     format = %-9v %{yyyy-MM-dd HH:mm:ss Z}d
     [AutoPrereqs]
     [OurPkgVersion]
     [MetaJSON]
     
     [@Git]
     allow_dirty = dist.ini
     allow_dirty = Changes
     allow_dirty = README.md
     
     [Author::Plicease::Resources]
     [InstallGuide]
     [MinimumPerl]
     [ConfirmRelease] 
     
     [ReadmeAnyFromPod]
     type     = text
     filename = README
     location = build
     
     [ReadmeAnyFromPod / ReadMePodInRoot]
     type     = markdown
     filename = README.md
     location = root
     
     [Author::Plicease::MarkDownCleanup]
     [Author::Plicease::Recommend]
     
     [Prereqs / NeedTestMore094]
     ; for subtest
     -phase     = test
     Test::More = 0.94
     
     [SpecialPrereqs]
     [CPANFile]

    Some exceptions:

    Perl 5.8

      [@Git] does not support Perl 5.8, so it is not a prereq there, and it
      isn't included in the bundle. As a result releasing from Perl 5.8 is
      not allowed.

    MSWin32

      Installing [@Git] on MSWin32 is a pain so it is also not a prereq on
      that platform, isn't used and as a result releasing from MSWin32 is
      not allowed.

OPTIONS

 installer

    Specify an alternative to [MakeMaker] ([ModuleBuild],
    [ModuleBuildTiny], or [ModuleBuildDatabase] for example).

    If installer is Alien, then any options with the alien_ prefix will be
    passed to Alien (minus the alien_ prefix).

    If installer is ModuleBuild, then any options with the mb_ prefix will
    be passed to ModuleBuild (including the mb_ prefix).

    If you have a inc/My/ModuleBuild.pm file in your dist, then this plugin
    bundle will assume installer is ModuleBuild and mb_class =
    My::ModuleBuild.

 readme_from

    Which file to pull from for the Readme (must be POD format). If not
    specified, then the main module will be used.

 release_tests

    If set to true, then include release tests when building.

 release_tests_skip

    Passed into the Author::Plicease::Tests if release_tests is true.

 travis_status

    if set to true, then include a link to the travis build page in the
    readme.

 mb_class

    if builder = ModuleBuild, this is the mb_class passed into the
    [ModuleBuild] plugin.

 github_repo

    Set the GitHub repo name to something other than the dist name.

 github_user

    Set the GitHub user name.

 copy_mb

    Copy Build.PL and cpanfile from the build into the git repository.
    Exclude them from gather.

    This allows other developers to use the dist from the git checkout,
    without needing to install Dist::Zilla and
    Dist::Zilla::PluginBundle::Author::Plicease.

SEE ALSO

    Author::Plicease::Init, MintingProfile::Plicease

AUTHOR

    Graham Ollis <perl@cpan.org>

COPYRIGHT AND LICENSE

    This software is copyright (c) 2012 by Graham Ollis.

    This is free software; you can redistribute it and/or modify it under
    the same terms as the Perl 5 programming language system itself.



Changes
Revision history for Dist-Zilla-PluginBundle-Author-Plicease

1.62      2015-01-07 14:47:05 -0500
  - escape < and > in AUTHOR / thanks section

1.61      2014-12-30 13:56:48 -0500
  - Make CPANFile part of the default [@Author::Plicease]
  - added copy_mb option
  - simplified prereqs

1.60      2014-12-19 16:40:12 -0500
  - option for skipping common spelling release test

1.59      2014-10-29 08:00:04 -0400
  - added [Author::Plicease::InstallerPerlVersion]

1.58      2014-10-14 07:35:02 -0400
  - release tests update

1.57      2014-10-08 08:32:09 -0400
  - upped the requirements on several Dist::Zilla::Plugin::* and Test::* modules
  - silenced an annoying warning generated by some dists during diagnostic
  - use Test::Fixme 0.14 and warn option

1.56      2014-09-20 07:29:30 -0400
  - update default .travis.yml to use verbose testing

1.55      2014-09-20 07:25:48 -0400
  - update default .travis.yml to use 5.20, and dzil instead of prove

1.54      2014-07-31 09:30:24 -0400
  - check .travis.yml for Perl 5.19 or 5.20 as appropriate

1.53      2014-07-06 09:25:17 -0400
  - require more recent Role::Tiny where used

1.52      2014-06-13 11:21:43 -0400
  - added [Author::Plicease::SpecialPrereqs]

1.51      2014-06-13 08:10:39 -0400
  - add prereq Test::More 0.94 to all dists
    I use subtest a lot

1.50      2014-06-10 07:53:11 -0400
  - added [Author::Plicease::Resources] and am
    using it for [@Author::Plicease] now.  This was
    needed to allow the repo for Alien-Libarchive be
    named Alien-Libarchive2
  - fixed .travis.yml template

1.49      2014-05-22 07:45:40 -0400
  - make this distribution properly dynamic
  - include [ACPS::RPM] in bundle, if it is there
  - turn off (by default) unused_vars.t for old dists

1.48      2014-04-30 13:23:38 -0400
  - don't include developer prereqs in diagnostic

1.47      2014-04-16 02:26:37 -0400
  - update .travis.yml template to reflect my current usage
  - fixed tempalte release.yml

1.46      2014-04-10 13:27:07 -0400
  - add to standard diagnostic
  - assume installer = ModuleBuild and mb_class = My::ModuleBuild if inc/My/ModuleBuild.pm is found
  - whitelist some commonly used tech spellings to pod system spelling test
  - add unused_vars.t release test
  - ability to skip some release tests from xt/release/release.yml

1.45      2014-04-08 11:04:49 -0400
  - allow override of diagnostic reporter with [Test] diag = +/-module or [@ACPS] diag = +/-module

1.44      2014-02-21 16:02:07 -0500
  - diagnostic doesn't preload EV unless it is a suggested prereq

1.43      2013-12-31 12:28:13 -0500
  - look for inc/run for scripted phase hooks

1.42      2013-12-21 17:09:01 -0500
  - slim down the prereqs.

1.41      2013-12-14 06:15:47 -0500
  - Don't allow release on MSWin32
  - Don't allow release on Perl 5.8 (implemented, but not documented in 1.40)

1.40      2013-12-13 22:16:55 -0500
  - Support Perl 5.8

1.39      2013-12-12 17:53:49 -0500
  - new dists use ModuleBuild

1.38      2013-12-05 18:29:39 -0500
  - added undocumented non_native_release option for [@Author::Plicease]

1.37      2013-11-27 12:37:59 -0500
  - added release_tests_skip option for [@Author::Plicease]

1.36      2013-11-19 10:27:13 -0500
  - mb_class option on [@Author::Plicease] passes to [ModuleBuild]

1.35      2013-11-18 07:55:09 -0500
  - rm dep on [Alien]

1.34      2013-11-11 17:49:01 -0500
  - [@Author::Plicease] alien_build_command and alien_install_command be mvp

1.33      2013-11-07 12:03:21 -0500
  - update version test

1.32      2013-11-07 11:28:33 -0500
  - support for [Alien] as the builder

1.31      2013-10-16 03:34:17 -0400
  - use [Bootstrap::lib] and relatives for release more consistent
    with my other distros

1.30      2013-10-15 11:54:55 -0400
  - make xt/release/version.t less strict, so that test versions
    (eg. 0.01_01) are allowed
  - fix for the regex removing the version from README.md so that
    it works for test versions (eg. 0.01_01)

1.29      2013-10-14 14:57:46 -0400
  - add xt/release/version.t to skel and distro

1.28      2013-10-12 14:31:26 -0400
  - use require instead of use in diagnostic

1.27      2013-10-09 14:44:12 -0400
  - add readme_from option to [@Author::Plicease]
  - documentation

1.26      2013-09-27 18:17:16 -0400
  - rm dep on use bytes (that was bad)
  - create t/00_diag.t and t/00_diag.txt as part of build
  - add [Author::Plicease::Recommend] and use it by default

1.25      2013-09-18 15:56:41 -0400
  - add dep on [TextTabs]

1.24      2013-09-10 06:41:12 -0400
  - tweak the MarkDownCleanup module

1.23      2013-09-09 20:20:02 -0400
  - silence annoying warning

1.22      2013-09-07 11:02:39 -0400
  - remove extra space in markdown verbatim blocks

1.21      2013-09-07 04:26:43 -0400
  - strip the version out of the README.md

1.20      2013-09-06 15:24:02 -0400
  - add [Author::Plicease::MarkDownCleanup] to [@Author::Plicease]

1.19      2013-09-06 15:08:16 -0400
  - renamed [Author::Plicease::TravisStatus] [Author::Plicease::MarkDownCleanup]

1.18      2013-09-06 14:56:09 -0400
  - add [Author::Plicease::TravisStatus]

1.17      2013-08-27 12:33:01 -0400
  - update prereq plugins to latest versions

1.16      2013-08-11 16:42:31 -0400
  - tweak date format

1.15      2013-08-08 16:01:30 -0400
  - use CPAN::Changes::Spec format for new entries in Changes

1.14      2013-07-18 13:30:51 America/New_York
  - add cpan option for [Author::Plicease::Upload]
  - update init2 to reflect previous upload merge

1.12      2013-07-18 09:11:35 America/New_York
  - merge [Author::Plicease::UploadToCPAN] and [Author::Plicease::UploadToMatrix] into [Author::Plicease::Upload]

1.11      2013-07-10 07:03:39 EST5EDT
  - fix author section header in [Author::Plicease::Thanks]

1.10      2013-07-10 06:48:50 EST5EDT
  - add [Author::Plicease::Thanks]

1.09      2013-06-22 09:47:47 EST5EDT
  - add an installer option for [@Author::Plicease]

1.08      2013-06-21 21:54:48 EST5EDT
  - updated dep versions


Makefile.PL

use strict;
use warnings;

use 5.010000;

use ExtUtils::MakeMaker 6.30;

use File::ShareDir::Install;
install_share module => "Dist::Zilla::MintingProfile::Author::Plicease", "profiles";


my %WriteMakefileArgs = (
  "ABSTRACT" => "Dist::Zilla plugin bundle used by Plicease",
  "AUTHOR" => "Graham Ollis <perl\@cpan.org>",
  "BUILD_REQUIRES" => {},
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => "6.30",
    "File::ShareDir::Install" => "0.03"
  },
  "DISTNAME" => "Dist-Zilla-PluginBundle-Author-Plicease",
  "EXE_FILES" => [],
  "LICENSE" => "perl",
  "NAME" => "Dist::Zilla::PluginBundle::Author::Plicease",
  "PREREQ_PM" => {
    "Dist::Zilla" => 0,
    "Dist::Zilla::File::FromCode" => 0,
    "Dist::Zilla::File::InMemory" => 0,
    "Dist::Zilla::MintingProfile::Clustericious" => "0.02",
    "Dist::Zilla::Plugin::Alien" => 0,
    "Dist::Zilla::Plugin::AutoMetaResources" => 0,
    "Dist::Zilla::Plugin::AutoPrereqs" => 0,
    "Dist::Zilla::Plugin::CSS::Compressor" => "0.01",
    "Dist::Zilla::Plugin::Git" => 0,
    "Dist::Zilla::Plugin::InsertExample" => "0.02",
    "Dist::Zilla::Plugin::InstallGuide" => 0,
    "Dist::Zilla::Plugin::JavaScript::Minifier" => "0.01",
    "Dist::Zilla::Plugin::MinimumPerl" => 0,
    "Dist::Zilla::Plugin::ModuleBuildDatabase" => "0.05",
    "Dist::Zilla::Plugin::NextRelease" => 0,
    "Dist::Zilla::Plugin::OurPkgVersion" => 0,
    "Dist::Zilla::Plugin::PodWeaver" => 0,
    "Dist::Zilla::Plugin::ReadmeAnyFromPod" => 0,
    "Dist::Zilla::Plugin::Run::BeforeBuild" => 0,
    "Dist::Zilla::Plugin::Template::Tiny" => "0.03",
    "Dist::Zilla::Plugin::TextTabs" => "0.02",
    "Dist::Zilla::Plugin::TwitterBootstrap" => "0.04",
    "Dist::Zilla::Plugin::UploadToCPAN" => 0,
    "Dist::Zilla::Plugin::jQuery" => "0.01",
    "Dist::Zilla::PluginBundle::ACPS" => "0.27",
    "Dist::Zilla::PluginBundle::Git" => 0,
    "Dist::Zilla::Role::AfterBuild" => 0,
    "Dist::Zilla::Role::AfterMint" => 0,
    "Dist::Zilla::Role::BeforeBuild" => 0,
    "Dist::Zilla::Role::FileFinderUser" => 0,
    "Dist::Zilla::Role::FileGatherer" => 0,
    "Dist::Zilla::Role::FileMunger" => 0,
    "Dist::Zilla::Role::InstallTool" => 0,
    "Dist::Zilla::Role::MetaProvider" => 0,
    "Dist::Zilla::Role::MintingProfile::ShareDir" => 0,
    "Dist::Zilla::Role::ModuleMaker" => 0,
    "Dist::Zilla::Role::PluginBundle::Easy" => 0,
    "Dist::Zilla::Role::PrereqSource" => 0,
    "Dist::Zilla::Role::TestRunner" => 0,
    "Encode" => 0,
    "File::Path" => 0,
    "File::chdir" => 0,
    "IPC::System::Simple" => 0,
    "JSON" => 0,
    "List::AllUtils" => 0,
    "Moose" => 0,
    "Path::Class" => "0.31",
    "PerlX::Maybe" => 0,
    "YAML::XS" => 0,
    "autodie" => 0,
    "namespace::autoclean" => 0,
    "strict" => 0,
    "utf8" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "Test::DZil" => 0,
    "Test::More" => 0
  },
  "VERSION" => "1.34",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


my %FallbackPrereqs = (
  "Dist::Zilla" => 0,
  "Dist::Zilla::File::FromCode" => 0,
  "Dist::Zilla::File::InMemory" => 0,
  "Dist::Zilla::MintingProfile::Clustericious" => "0.02",
  "Dist::Zilla::Plugin::Alien" => 0,
  "Dist::Zilla::Plugin::AutoMetaResources" => 0,
  "Dist::Zilla::Plugin::AutoPrereqs" => 0,
  "Dist::Zilla::Plugin::CSS::Compressor" => "0.01",
  "Dist::Zilla::Plugin::Git" => 0,
  "Dist::Zilla::Plugin::InsertExample" => "0.02",
  "Dist::Zilla::Plugin::InstallGuide" => 0,
  "Dist::Zilla::Plugin::JavaScript::Minifier" => "0.01",
  "Dist::Zilla::Plugin::MinimumPerl" => 0,
  "Dist::Zilla::Plugin::ModuleBuildDatabase" => "0.05",
  "Dist::Zilla::Plugin::NextRelease" => 0,
  "Dist::Zilla::Plugin::OurPkgVersion" => 0,
  "Dist::Zilla::Plugin::PodWeaver" => 0,
  "Dist::Zilla::Plugin::ReadmeAnyFromPod" => 0,
  "Dist::Zilla::Plugin::Run::BeforeBuild" => 0,
  "Dist::Zilla::Plugin::Template::Tiny" => "0.03",
  "Dist::Zilla::Plugin::TextTabs" => "0.02",
  "Dist::Zilla::Plugin::TwitterBootstrap" => "0.04",
  "Dist::Zilla::Plugin::UploadToCPAN" => 0,
  "Dist::Zilla::Plugin::jQuery" => "0.01",
  "Dist::Zilla::PluginBundle::ACPS" => "0.27",
  "Dist::Zilla::PluginBundle::Git" => 0,
  "Dist::Zilla::Role::AfterBuild" => 0,
  "Dist::Zilla::Role::AfterMint" => 0,
  "Dist::Zilla::Role::BeforeBuild" => 0,
  "Dist::Zilla::Role::FileFinderUser" => 0,
  "Dist::Zilla::Role::FileGatherer" => 0,
  "Dist::Zilla::Role::FileMunger" => 0,
  "Dist::Zilla::Role::InstallTool" => 0,
  "Dist::Zilla::Role::MetaProvider" => 0,
  "Dist::Zilla::Role::MintingProfile::ShareDir" => 0,
  "Dist::Zilla::Role::ModuleMaker" => 0,
  "Dist::Zilla::Role::PluginBundle::Easy" => 0,
  "Dist::Zilla::Role::PrereqSource" => 0,
  "Dist::Zilla::Role::TestRunner" => 0,
  "Encode" => 0,
  "File::Path" => 0,
  "File::chdir" => 0,
  "IPC::System::Simple" => 0,
  "JSON" => 0,
  "List::AllUtils" => 0,
  "Moose" => 0,
  "Path::Class" => "0.31",
  "PerlX::Maybe" => 0,
  "Test::DZil" => 0,
  "Test::More" => 0,
  "YAML::XS" => 0,
  "autodie" => 0,
  "namespace::autoclean" => 0,
  "strict" => 0,
  "utf8" => 0,
  "warnings" => 0
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);

{
package
MY;
use File::ShareDir::Install qw(postamble);
}




Build.PL

# This file was automatically generated by Dist::Zilla::Plugin::ModuleBuild v5.029.
use strict;
use warnings;

use Module::Build 0.3601;
use lib qw{inc}; use My::ModuleBuild;

my %module_build_args = (
  "build_requires" => {
    "Module::Build" => "0.3601"
  },
  "configure_requires" => {
    "Module::Build" => "0.3601",
    "perl" => "5.006"
  },
  "dist_abstract" => "Dist::Zilla plugin bundle used by Plicease",
  "dist_author" => [
    "Graham Ollis <perl\@cpan.org>"
  ],
  "dist_name" => "Dist-Zilla-PluginBundle-Author-Plicease",
  "dist_version" => "1.62",
  "license" => "perl",
  "module_name" => "Dist::Zilla::PluginBundle::Author::Plicease",
  "recommends" => {
    "PerlX::Maybe::XS" => 0,
    "YAML::XS" => 0
  },
  "recursive_test_files" => 1,
  "requires" => {
    "Dist::Zilla" => "5.020",
    "Dist::Zilla::Plugin::AutoMetaResources" => "1.20",
    "Dist::Zilla::Plugin::CopyFilesFromBuild" => 0,
    "Dist::Zilla::Plugin::InstallGuide" => "1.200003",
    "Dist::Zilla::Plugin::MinimumPerl" => "1.003",
    "Dist::Zilla::Plugin::OurPkgVersion" => "0.005001",
    "Dist::Zilla::Plugin::PodWeaver" => "4.006",
    "Dist::Zilla::Plugin::ReadmeAnyFromPod" => "0.142470",
    "Dist::Zilla::Plugin::Run::BeforeBuild" => "0.026",
    "File::chdir" => 0,
    "IPC::System::Simple" => "1.25",
    "JSON::PP" => 0,
    "Moose" => 0,
    "Path::Class" => "0.31",
    "PerlX::Maybe" => "0.003",
    "Test::Fixme" => "0.14",
    "Test::Pod" => "1.44",
    "Test::Pod::Coverage" => "1.08",
    "Test::Version" => "1.002001",
    "YAML" => 0,
    "autodie" => 0,
    "namespace::autoclean" => 0,
    "perl" => "5.008001"
  },
  "script_files" => [],
  "share_dir" => {
    "module" => {
      "Dist::Zilla::MintingProfile::Author::Plicease" => "profiles"
    }
  },
  "test_requires" => {
    "Test::More" => "0.94",
    "perl" => "5.008001"
  }
);


my %fallback_build_requires = (
  "Module::Build" => "0.3601",
  "Test::More" => "0.94",
  "perl" => "5.008001"
);


unless ( eval { Module::Build->VERSION(0.4004) } ) {
  delete $module_build_args{test_requires};
  $module_build_args{build_requires} = \%fallback_build_requires;
}

my $build = My::ModuleBuild->new(%module_build_args);

$build->create_build_script;
